pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -149,7 +149,14 @@ func (pc *PodCache) onEvent(old, pod *v1.Pod, ev model.Event) error {
 	// PodIP will be empty when pod is just created, but before the IP is assigned
 	// via UpdateStatus.
 	if len(ip) == 0 {
-		return nil
+		// However, in the case of an Eviction, the event that marks the pod as Failed may *also* have removed the IP.
+		// If the pod *used to* have an IP, then we need to actually delete it.
```

Comment:
Talking to myself...

So the event may be "update" but then we hit `deleteIP` anyways due to shouldPodBeInEndpoints and translate to a delete.

deleteIP already handles the case of "IP was removed" since its a map of ip->[]pods.

We then fallthrough to notifyWorkloadHandlers, build an EP with no IP but WorkloadInstance is identified by name not IP. This looks good :heavy_check_mark: